sensing/src/main/java/com/google/android/sensing/impl/SensingEngineImpl.kt

```diff
@@ -178,6 +181,34 @@ internal class SensingEngineImpl(
     TODO("Not yet implemented")
   }
 
+  override suspend fun getCaptureInfo(captureId: String): CaptureInfo {
+    return database.getCaptureInfo(captureId)
+  }
+
+  override suspend fun deleteDataInCapture(captureId: String): Boolean {
+    val captureInfo =
+      try {
+        getCaptureInfo(captureId)
+      } catch (e: ResourceNotFoundException) {
+        null
+      } ?: return true
+
+    // Step 1: Delete db records
+    database.deleteRecordsInCapture(captureId)
```

Comment:
In order to avoid inconsistencies in the data, like lets says table records are deleted but Files not being deleted (due to failures/restarts), can we make this complete transaction more robust? 

Also, how about the Zipped records , if they are not yet synced? I am assuming they wil be still be available.

Reply:
As we discussed, one way to do this is 
- First soft delete the records
- Delete the files
- Hard delete the records.
This way even if the transaction fails in between, retriggering the API should take care of complete deletion.

Reply:
You are right. I have created this [issue](https://github.com/google-research/CVD-paper-mobile-camera-example/issues/44)
As for zipped files, they get deleted too as their location is in parallel to the data (reference by ResourceInfo), ie, inside the captureFolder.

Reply:
Thanks for the clarification